File: spot/clients/rest-http/node/run.example.js
require('dotenv').config();
const SpotApiClient = require('./client');

async function main() {
  try {
    // Choose which client to use
    // You can use either JWT or API Key authentication based on your credentials
    let client;

    // Initialize client with JWT authentication
    if (process.env.JWT_TOKEN) {
      console.log('Initializing client with JWT authentication...');
      client = new SpotApiClient({
        jwt: process.env.JWT_TOKEN,
        timeout: 20_000 // optional field to configure req timeout - default=30_000 ms
      });
    } else if (process.env.API_KEY && process.env.SECRET_KEY) {
      // Initialize client with API Key authentication
      console.log('Initializing client with API Key authentication...');
      client = new SpotApiClient({
        apiKey: process.env.API_KEY,
        secretKey: process.env.SECRET_KEY,
        timeout: 20_000 // optional field to configure req timeout - default=30_000 ms
      });
    } else {
      // Throw error if credentials are insufficient
      throw new Error('Credentials Insufficient. One of either JWT or API Key & Secret Key is required');
    }

    // System APIs
    // console.log('\n=== System APIs ===');
    // console.log('Fetching system status...');
    // const status = await client.getServiceStatus();
    // console.log('System status:', status);

    // console.log('\nFetching server time...');
    // const serverTime = await client.getServerTime();
    // console.log('Server time:', serverTime);

    // // Market Data APIs
    // console.log('\n=== Market Data APIs ===');
    // console.log('Fetching all tickers...');
    // const allTickers = await client.getAllTickers();
    // console.log('All tickers:', allTickers);

    // console.log('\nFetching BTC-INR ticker...');
    // const ticker = await client.getTicker('BTC-INR');
    // console.log('BTC-INR ticker:', ticker);

    // console.log('\nFetching BTC-INR order book...');
    // const orderBook = await client.getOrderBook('BTC-INR');
    // console.log('BTC-INR order book:', orderBook);

    // console.log('\nFetching BTC-INR order book ticker...');
    // const orderBookTicker = await client.getOrderBookTicker('BTC-INR');
    // console.log('BTC-INR order book ticker:', orderBookTicker);

    // console.log('\nFetching BTC-INR recent trades...');
    // const recentTrades = await client.getRecentTrades('BTC-INR');
    // console.log('BTC-INR recent trades:', recentTrades);

    // console.log('\nFetching BTC-INR kline data...');
    // const kline = await client.getKline('BTC-INR', '1h', Date.now() - 24 * 60 * 60 * 1000, Date.now());
    // console.log('BTC-INR kline data:', kline);

    // // Exchange APIs
    // console.log('\n=== Exchange APIs ===');
    // console.log('Fetching trading pairs...');
    // const tradingPairs = await client.getTradingPairs();
    // console.log('Trading pairs:', tradingPairs);

    // console.log('\nFetching coin settings...');
    // const coinSettings = await client.getCoinSettings();
    // console.log('Coin settings:', coinSettings);

    // Private APIs (require authentication)
    try {
      console.log('\n=== Private APIs ===');
      console.log('Fetching account balance...');
      const balance = await client.getAccountBalance();
      console.log('Account balance:', balance);

      console.log('\nFetching BTC-INR exchange fee...');
      const exchangeFee = await client.getExchangeFee('BTC-INR', 'BUY');
      console.log('BTC-INR exchange fee:', exchangeFee);

      console.log('\nFetching open orders...');
      const openOrders = await client.getOrders('BTC-INR', 'ACTIVE');
      console.log('Open orders:', openOrders);

      console.log('\nFetching filled orders...');
      const filledOrders = await client.getOrders('BTC-INR', 'FILLED');
      console.log('Filled orders:', filledOrders);

      console.log('\nFetching cancelled orders...');
      const cancelledOrders = await client.getOrders('BTC-INR', 'CANCELLED');
      console.log('Cancelled orders:', cancelledOrders);

      // Example: Place a market order (commented out for safety)
      console.log('\nPlacing a market order...');
      const marketOrder = {
        symbol: 'BTC-INR',
        side: 'BUY',
        type: 'MARKET',
        quoteOrderQty: '100'
      };
      const marketOrderResult = await client.placeOrder(marketOrder);
      console.log('Market order placed:', marketOrderResult);

      console.log('\nPlacing a limit order...');
      const limitOrder = {
        symbol: 'BTC-INR',
        side: 'BUY',
        type: 'LIMIT',
        quantity: '0.0001',
        price: '5500000' 
      };
      const limitOrderResult = await client.placeOrder(limitOrder);
      console.log('Limit order placed:', limitOrderResult); 

      // Example: Get order details (commented out as it requires a valid order ID)
      console.log('\nFetching order details...');
      const orderDetails = await client.getOrderDetails('5987589');
      console.log('Order details:', orderDetails);

      // Example: Get order fills (commented out as it requires a valid order ID)
      console.log('\nFetching order fills...');
      const orderFills = await client.getOrderFills('5987589');
      console.log('Order fills:', orderFills);

      // Example: Cancel order (commented out as it requires a valid order ID)
      console.log('\nCanceling order...');
      const cancelResult = await client.cancelOrder('5987589');
      console.log('Cancel result:', cancelResult);

      // Example: Cancel all orders (commented out for safety)
      console.log('\nCanceling all orders...');
      const cancelAllResult = await client.cancelAllOrders('BTC-INR');
      console.log('Cancel all result:', cancelAllResult);

    } catch (error) {
      console.error('Private endpoint error:', error.message);
      console.log('This is likely due to insufficient credentials. Please check your .env file.');
    }
  } catch (error) {
    console.error('Error:', error.message);
  }
}

main(); 

File: futures/clients/rest-http/python/run_example.py
"""
Example usage of the Futures API Client
"""

import os
from dotenv import load_dotenv
from client.client import FuturesApiClient


def run_examples():
    """
    Simple example demonstrating how to use the Futures API client
    """
    try:
        # Load environment variables from .env file
        load_dotenv()

        # Choose which client to use for the examples
        # You can use either JWT or API Key authentication based on your credentials
        client = None

        # Initialize client with JWT authentication
        if os.getenv("JWT_TOKEN"):
            print("Initializing client with JWT authentication...")
            client = FuturesApiClient(
                jwt=os.getenv("JWT_TOKEN"),
                timeout=20  # optional field to configure req timeout - default=30 secs
            )
        elif os.getenv("API_KEY") and os.getenv("SECRET_KEY"):
            # Initialize client with API Key authentication
            print("Initializing client with API Key authentication...")
            client = FuturesApiClient(
                api_key=os.getenv("API_KEY"),
                secret_key=os.getenv("SECRET_KEY"),
                timeout=20  # optional field to configure req timeout - default=30 secs
            )
        else:
            # Throw error if credentials are insufficient
            raise ValueError("Credentials Insufficient. One of either JWT or Api Key & Secret Key is required")

        # Example 1: Call a public endpoint (no authentication required)
        print("\nExample 1: Fetching system status (public endpoint)")
        status = client.get_system_status()
        print(f"System status: {status}")

        # Example 2: Call another public endpoint with parameters
        print("\nExample 2: Fetching order book for BTCUSDT (public endpoint)")
        order_book = client.get_order_book("BTCUSDT")
        print("Order book sample:", {
            "symbol": order_book.get("data", {}).get("symbol"),
            "bidCount": len(order_book.get("data", {}).get("bids", [])),
            "askCount": len(order_book.get("data", {}).get("asks", []))
        })

        # Example 3: Call a private endpoint (requires authentication)
        print("\nExample 3: Fetching account balance (private endpoint)")
        balance = client.get_balance()
        print(f"Balance: {balance.get('data')}")

        # Example 4: Place a market order (private endpoint)
        print("\nExample 4: Creating a market order (private endpoint)")
        # Uncomment to execute the order
        """
        order = await client.create_order({
            "symbol": "BTCUSDT",
            "amount": 0.002,  # Small amount for testing
            "side": "BUY",
            "type": "MARKET",
            "marginAsset": "USDT"
        })
        print(f"Order created: {order.get('data')}")
        """
        print("Order creation example (commented out for safety)")

    except Exception as error:
        print(f"Error: {str(error)}")


if __name__ == "__main__":
    # Run the examples
    run_examples()
